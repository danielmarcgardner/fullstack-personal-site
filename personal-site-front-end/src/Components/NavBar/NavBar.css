#navbar {
  width: 101%;
  height: 50px;
  background-color: #202938;
}

.navitem {
  display: inline-block;
  color: #679ffc;
  text-align: center;
  height: 35px;
  padding-top:15px;
  width: 75px;
}

.navitem:hover {
  color: #202938;
  background-color: #679ffc;
  cursor: pointer
}

#navlinks {
  float: right;
  margin-right: 5%;
  height:inherit;
}
