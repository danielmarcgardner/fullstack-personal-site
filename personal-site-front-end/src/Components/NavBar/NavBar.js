import React from 'react';
import renderIf from 'render-if';
import './NavBar.css';

const navbar = ({ user, history, signOutUser }) => (
  <div id="navbar">
    {renderIf(user.loggedIn === false)(
      <div className="navsignin">
        <div className="navitem" onClick={() => history.push('/signin')}>
          Sign In
        </div>
        <div className="navitem" onClick={() => history.push('/signup')}>
          Sign Up
        </div>
      </div>,
    )}
    {
      renderIf(user.loggedIn === true)(
        <div className="navsignin">
          <div id="usernamedisplay" onClick={() => signOutUser()}>
            {`${user.name}`} <br />
            Sign Out
          </div>
        </div>,
      )
    }
    <div id="navlinks">
      <div className="navitem" onClick={() => history.push('/')}>
        Home
      </div>
      <div className="navitem" onClick={() => history.push('/projects')}>
        Projects
      </div>
      <div className="navitem" onClick={() => history.push('/blog')}>
        Blog
      </div>
    </div>
  </div>
);

export default navbar;
