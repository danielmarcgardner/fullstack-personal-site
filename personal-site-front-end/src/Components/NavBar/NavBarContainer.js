import { connect } from 'react-redux';
import NavBar from './NavBar';

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(NavBar);
