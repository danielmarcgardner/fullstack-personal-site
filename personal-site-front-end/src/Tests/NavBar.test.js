import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../Components/NavBar/NavBar';

const component = shallow(<NavBar />);

describe('NavBar', () => {
  it('Renders', () => {
    expect(component).toMatchSnapshot();
  });
  it('Has 4 navitems', () => {
    expect(component.find('.navitem').length).toEqual(4);
  });
});
