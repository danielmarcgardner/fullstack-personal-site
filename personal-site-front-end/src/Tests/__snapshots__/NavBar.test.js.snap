// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    id="navbar"
>
    <div
        className="navitem"
    >
        Sign In
    </div>
    <div
        id="navlinks"
    >
        <div
            className="navitem"
            onClick={[Function]}
        >
            Home
        </div>
        <div
            className="navitem"
            onClick={[Function]}
        >
            Projects
        </div>
        <div
            className="navitem"
            onClick={[Function]}
        >
            Blog
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      id="navbar"
>
      <div
            className="navitem"
      >
            Sign In
      </div>
      <div
            id="navlinks"
      >
            <div
                  className="navitem"
                  onClick={[Function]}
            >
                  Home
            </div>
            <div
                  className="navitem"
                  onClick={[Function]}
            >
                  Projects
            </div>
            <div
                  className="navitem"
                  onClick={[Function]}
            >
                  Blog
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <navbar />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          id="navbar"
>
          <div
                    className="navitem"
          >
                    Sign In
          </div>
          <div
                    id="navlinks"
          >
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Home
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Projects
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Blog
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          id="navbar"
>
          <div
                    className="navitem"
          >
                    Sign In
          </div>
          <div
                    id="navlinks"
          >
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Home
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Projects
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Blog
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <navbar />,
}
`;
